main.bal

```diff
@@ -1,56 +1,101 @@
 import ballerina/io;
 import ballerina/log;
-// import ballerina/test;
 import ballerina/tcp;
-// import ballerina/lang.'string;
-import ballerina/regex;
 
-
-//bind the service to the port
 service on new tcp:Listener(3000) {
 
-    remote function onConnect(tcp:Caller caller)
-                              returns tcp:ConnectionService {
+    remote function onConnect(tcp:Caller caller) 
+                            returns tcp:ConnectionService {
         io:println("Client connected to the server: ", caller.remotePort);
         return new Service();
     }
 }
 
 service class Service {
-//once the content is received from the client, this method is invoked
 
     remote function onBytes(tcp:Caller caller, readonly & byte[] data) 
         returns tcp:Error? {
-        // io:println(string:fromBytes(data));  
+        error? e = self.handleInput(caller, data);
+
+        if e is error {
+            return;
+        }
+    }
+
+    byte[] buffer = [];
+    int? expectedLen = ();
+    string messageContent = "";
```

Comment:
can this be a local variable as well?

Reply:
the messages will not pass if we make it a local variable